### PR TITLE
[GSoC] Kotlin migration TextImporter.java

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Purpose / Description

Kotlin Migration

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt [hunk 1]
```diff
@@ -1,313 +1,263 @@
...
+        return mNumFields
     }
 
-
-    private ForeignNote noteFromFields(List<String> fields) {
-        ForeignNote note = new ForeignNote();
-        note.mFields.addAll(fields);
-        note.mTags.addAll(Arrays.asList(mTagsToAdd));
-        return note;
+    private fun noteFromFields(fields: List<String?>): ForeignNote {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:94]: Add a KotlinCleanup for this function: simply fun with scope function.

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt [hunk 2]
```diff
@@ -1,313 +1,263 @@
...
+    private fun openFile() {
+        mDialect = null
+        mFileObj = FileObj.open(mFile)
+        val firstLine = firstFileLine.orElse(null)
         if (firstLine != null) {
             if (firstLine.startsWith("tags:")) {
-                String tags = firstLine.substring("tags:".length()).trim();
-                mTagsToAdd = tags.split(" ");
-                this.mFirstLineWasTags = true;
+                val tags = firstLine.substring("tags:".length).trim { it <= ' ' }
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:121]: Did the tool used in the conversion the trim(predicate) method or did you added it yourself?

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt (hunk 2 above)
Comment (ShridharGoel) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:121]: Found this while searching about it: https://stackoverflow.com/a/45652573

Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:121]: @ShridharGoel Thank you, I did a quick migration of the file and saw that indeed the tool used the trim(predicate) method.

Comment (ShridharGoel) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:121]: @lukstbit Welcome.

**Review by lukstbit — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt [hunk 3]
```diff
@@ -1,313 +1,264 @@
...
+        return mNumFields
     }
 
-
-    private ForeignNote noteFromFields(List<String> fields) {
-        ForeignNote note = new ForeignNote();
-        note.mFields.addAll(fields);
-        note.mTags.addAll(Arrays.asList(mTagsToAdd));
-        return note;
+    @KotlinCleanup("simply fun with scope function")
```
Comment (ShridharGoel) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:94]: Should have been "simplify".

Comment (divyansh-dxn) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:94]: It's a temporary annotation so I don't think it's that important as long as it's understandable.

**Pull request merged**